lib/semian.rb

```diff
@@ -143,7 +148,7 @@ def register(name, tickets:, permissions: 0660, timeout: 0, error_threshold:, er
       exceptions: Array(exceptions) + [::Semian::BaseError],
       implementation: ::Semian::Simple,
     )
-    resource = Resource.new(name, tickets: tickets, permissions: permissions, timeout: timeout)
+    resource = Resource.instance(name, tickets: tickets, quota: quota, permissions: permissions, timeout: timeout)
```

Comment:
We now call resource.instance instead of resource.new directly, to ensure that only one resource can be registered per name. This is essential to ensure for calculating the tickets from quota accurately.